app/Libraries/Markdown/OsuMarkdown.php

```diff
@@ -203,23 +267,94 @@ private function getIndexableConverter(): MarkdownConverter
         return $this->indexableConverter;
     }
 
-    private function createBaseEnvironment(): ConfigurableEnvironmentInterface
+    private function createBaseEnvironment(bool $isHtml = false): Environment
     {
-        $environment = Environment::createCommonMarkEnvironment()
-            ->addExtension(new AutolinkExtension())
-            ->addExtension(new TableExtension());
+        $config = $this->commonmarkConfig;
+
+        if ($isHtml) {
+            $config = array_merge(
+                $config,
+                [
+                    'osu_extension' => $this->osuExtensionConfig,
+                    'default_attributes' => $this->createDefaultAttributesConfig(),
+                ]
+            );
+        }
+
+        $environment = new Environment($config);
```

Comment:
the additional config and the `DefaultAttributesExtension` below belong in `getHtmlConverterAndExtension`

```
$environment = new Environment(array_merge($this->commonmarkConfig, $extraConfig));
```